python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -104,22 +107,45 @@ def _parse_op_fn(op: AbstractUDFMap):
         fn_constructor_args = op._fn_constructor_args or ()
         fn_constructor_kwargs = op._fn_constructor_kwargs or {}
 
-        op_fn = make_callable_class_concurrent(op_fn)
+        if inspect.isasyncgenfunction(op._fn.__call__):
```

Comment:
Add a TODO that we should 1) support non-generator async functions; 2) make the entire map actor async.